### PR TITLE
Added bindings for IControlMode2 class and Matlab/C++ vectors conversion

### PR DESCRIPTION
**Pull request by nunoguedelha**
This change adds the support of bindings to the `IControlMode2` interface methods. It extends:
- the `yarp::dev::PolyDriver` class adding a wrapping of `viewIControlMode2()` which casts the device driver to the `IControlMode2` interface
- the `yarp::dev::IControlMode2` interface to deal with parameters that are not correctly translated by **SWIG**
- the `std::vector<int>` class for adding the methods `toMatlab()` and `fromMatlab()` converting `std::vector<int>` objects to Matlab `double` arrays and vice-versa.

### CODE REVIEW

File: bindings/yarp.i [hunk 1]
```diff
@@ -463,7 +464,10 @@ MAKE_COMMS(Bottle)
   %template(DVector) std::vector<double>;
   %template(BVector) std::vector<bool>;
   %template(SVector) std::vector<std::string>;
+  // Extend IVector for handling conversion of vectors from and to Matlab
+  %include "IVector_fromTo_matlab.i"
```
Comment (traversaro) [inline bindings/yarp.i]: This should be included only when generating the code for matlab (mex-based) bindings, so perhaps it should be guarded by an `#ifdef SWIGMATLAB` clause? I wonder how it is compiling fine on Travis.

**Comment by drdanz** [comment]
@nunoguedelha Is this PR finished or abandoned? if it is finished, can you please rebase it and remove the WIP from the title?
Also I think this might cause some conflict with #1372

**Comment by drdanz** [comment]
Also, since this is a new feature, it should be in **devel**, not in **master**

File: bindings/IVector_fromTo_matlab.i [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        // create Matlab vector and map it to pointer 'd'
+        size_t selfDim = self->size();
+        mxArray *p  = mxCreateDoubleMatrix(selfDim, 1, mxREAL);
+        double* d = static_cast<double*>(mxGetData(p));
+
+        // copy items from 'self' to 'd'
+        const int* selfData = self->data();
+        for(size_t i=0; i<selfDim; i++)
+        {
+            d[i] = selfData[i];
```
Comment (traversaro) [inline bindings/IVector_fromTo_matlab.i]: Nitpick: just for clarity it would be good to have a static_cast<double>() here.

**Comment by traversaro** [comment]
Good for me. @drdanz it is not clear to me what Codacy is complaining about.

**Comment by traversaro** [comment]
This PR will be merge after https://github.com/robotology/yarp/pull/1372 .

**Comment by nunoguedelha** [comment]
@traversaro @Nicogene , rebased and added the static cast after comment https://github.com/robotology/yarp/pull/1311#pullrequestreview-71771832.

**Comment by nunoguedelha** [comment]
> After #1372, all files specific for one binding language were moved in a subdirectory, please move IVector_fromTo_matlab.i to a matlab subfolder...

@drdanz  Done.

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**